src/main/java/keycontacts/model/student/Group.java

```diff
@@ -87,4 +87,16 @@ public int hashCode() {
     public String toString() {
         return groupName;
     }
+
+    @Override
+    public int compareTo(Group o) {
+        if (isNoGroup() && o.isNoGroup()) {
```

Comment:
although it's not a part of this commit, I think we could consider renaming isNoGroup -> hasNoGroup. Just a small detail.

Reply:
Reason is because its a method of the group (not a method of the student)